### PR TITLE
feat(streaming): support sum0 to fix stream 2 phase count output with empty input rows.

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Support `Sum0` with initial value zero. (Default `Sum` with initial value NULL)
- Fix `array_agg` output

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
#5994 #5962

### CODE REVIEW

File: src/stream/src/executor/aggregation/agg_impl/foldable.rs [hunk 1]
```diff
@@ -127,6 +127,41 @@ where
     }
 }
 
+/// `I64Sum0` sums two primitives by `accumulate` and `retract` functions.
+/// It produces the same type of output as input `S`.
+#[derive(Debug)]
+pub struct I64Sum0 {}
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/agg_impl/foldable.rs:133]: How about storing a `PrimitiveSummable<i64, i64>` and forwarding the `accumulate` and `retract` to it?

Comment (chenzl25) [reply src/stream/src/executor/aggregation/agg_impl/foldable.rs:133]: Sounds good.

File: src/stream/src/executor/aggregation/state_cache/array_agg.rs [hunk 2]
```diff
@@ -43,11 +46,7 @@ mod tests {
         let agg = ArrayAgg;
 
         let mut cache = OrderedCache::new(10);
-        // FIXME(yuchao): the behavior is not compatible with PG, #5962
-        assert_eq!(
-            agg.aggregate(cache.iter_values()),
-            Some(ListValue::new(vec![]).into())
-        );
+        assert_eq!(agg.aggregate(cache.iter_values()), None,);
```
Comment (richardchien) [inline src/stream/src/executor/aggregation/state_cache/array_agg.rs:49]:
```suggestion
        assert_eq!(agg.aggregate(cache.iter_values()), None);
```

File: src/frontend/src/expr/agg_call.rs [hunk 3]
```diff
@@ -92,6 +92,12 @@ impl AggCall {
             },
             (AggKind::Sum, _) => return invalid(),
 
+            (AggKind::Sum0, [input]) => match input {
+                DataType::Int64 => DataType::Int64,
+                _ => return invalid(),
+            },
```
Comment (richardchien) [inline src/frontend/src/expr/agg_call.rs:98]:
```suggestion
            (AggKind::Sum0, [DataType::Int64]) => DataType::Int64,
```

File: src/frontend/src/optimizer/plan_node/generic.rs [hunk 4]
```diff
@@ -293,7 +293,8 @@ impl PlanAggCall {
     pub fn partial_to_total_agg_call(&self, partial_output_idx: usize) -> PlanAggCall {
         let total_agg_kind = match &self.agg_kind {
             AggKind::Min | AggKind::Max | AggKind::StringAgg | AggKind::FirstValue => self.agg_kind,
-            AggKind::Count | AggKind::Sum | AggKind::ApproxCountDistinct => AggKind::Sum,
+            AggKind::Count | AggKind::Sum0 => AggKind::Sum0,
+            AggKind::Sum | AggKind::ApproxCountDistinct => AggKind::Sum,
```
Comment (richardchien) [inline src/frontend/src/optimizer/plan_node/generic.rs:297]: Should `ApproxCountDistinct` also use `Sum0` as its final phase?

File: src/stream/src/executor/aggregation/agg_state.rs [hunk 5]
```diff
@@ -133,4 +133,15 @@ impl<S: StateStore> AggState<S> {
             }
         }
     }
+
+    pub fn reset(&mut self) {
```
Comment (richardchien) [inline src/stream/src/executor/aggregation/agg_state.rs:137]: May add comment to explain what this method does.

**Review by richardchien — APPROVED** [review]
LGTM!

File: src/frontend/src/optimizer/plan_node/generic.rs (hunk 4 above)
Comment (chenzl25) [reply src/frontend/src/optimizer/plan_node/generic.rs:297]: Not really sure. I think both are ok.

**Pull request merged**